src/index.ts

```diff
@@ -131,8 +216,20 @@ export interface RunOptions {
      * The default is false.
      */
     usePolling?: boolean;
+
+    /**
+     * A token used to authenticate with the service.
+     */
+    runToken?: string;
+
+    /**
+     * JSON containing info about the portal item including item type and service url.
+     */
+    portalItemInfo?: PortalItemResponse;
```

Comment:
The `portalItemInfo` is a bit of an awkward one for the caller. There is a bunch of irrelevant stuff on it because it is a response object. 

What if we just had a `serviceUrl?: string` option instead.

In `run()` we could only bother with the `isValidItemType` check when it calls `getPortalItemInfo`.

Reply:
I went back and forth on this a bit before.  Changed.